### DISCUSSION TITLE
Grab Captions from Disney+

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#695] [Share]
> 📝 Requires "Allow JavaScript From Apple Events" (see image at bottom)

https://user-images.githubusercontent.com/36073/150913892-7fb80f2d-8ea9-4835-ad60-2bfe39f9916b.mp4

[Open grab-caption in Script Kit](https://scriptkit.com/api/new?name=grab-caption&url=https://gist.githubusercontent.com/johnlindquist/b1d146e0308089002e2a48377af0e59d/raw/0697ca4ea5f67a5302279214b3dda002f9889a5c/grab-caption.ts")

```js
// Name: Grab Caption

import "@johnlindquist/kit"

// This query selector is specific to Disney plus
let js = `

Array.from(document.querySelectorAll('.dss-subtitle-renderer-line')).map(el => el.innerText)

`

let value = await applescript(`
tell application "Google Chrome" to tell window 1
	set str to execute active tab javascript "${js}"
	return str
end tell
`)

await div(md(`## ${value}`))

// To copy to clipboard, use:
// copy(value)
```

![image](https://user-images.githubusercontent.com/36073/150914006-5f64c3e1-194a-48aa-807e-1c8370044c1f.png)

**Comment by jsjoeio** [comment]
This is amazing!
